src/main/java/uk/ac/ebi/ampt2d/accession/file/FileAccessionRepository.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.file;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.ampt2d.accession.AccessionRepository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class FileAccessionRepository implements AccessionRepository<File, UUID> {
+
+    @Autowired
+    private FileCrudRepository fileJpaRepository;
+
+    @Override
+    public Map<File, UUID> get(List<File> objects) {
+        List<String> checksums = objects.stream().map(File::getHash).collect(Collectors.toList());
+        List<File> filesInRepository = fileJpaRepository.findByHashIn(checksums);
+        return filesInRepository.stream().collect(Collectors.toMap(Function.identity(), File::getAccession));
+    }
+
+    @Override
+    public void add(Map<File, UUID> accessions) {
```

Comment:
As previously discussed, the way the `File` class is declared and the interface of the repository are redundant. If a `File` contains its own accession then we are forcing it to be a UUID, and won't work with repositories declared with any other type. At the same time the repository interface could be simpler, because we don't need to declare the accession type, as it is implicit in the stored object.

On the other hand, I appreciate that the only use case this service needs to support is accessioning, so creating `File` and `FileAccession` classes could just add overhead. Maybe `File` could support customization of the accession type via generics? Would the serialization work?

Simplifying it that way could also allow us to use Spring Data REST for the repositories + web services layers.